### ISSUE TITLE
How to manage deletion of services and routes in distributed configuration

### ISSUE THREAD

**Issue by lays147** [Kong/deck#1236]
Hi all, 
In the scenario that I'm using several yaml files to configure my services and routes, when I deleted some of them, decK is unable to diff in a file that doesn't exist. 

If I remove the file with the service and routes, I get orphan resources on my gateway. 
If I keep the file and remove the service and routes, leaving only the `select_tags` block, decK doesn't get any diff. 
If I leave the file empty, decK diff plans the destruction of everything that is configured in the gateway.

So, how can one do this cleaning job without leaving orphan configuration in the gateway?

**Comment by gchristidis** [comment]
I use an empty file with the select tags and when that is sync'd Deck will remove all resources that have the selected tags and after removal deck diff reports no changes.

```yaml
_format_version: "1.1"
_info:
  select_tags:
    - tag1 
    - tag2
```

In the system i use it in it does mean that individual yaml file based on the combined tags can never be deleted from the system, deleting resources for those tags simply creates an empty file as above that always lives with the system

**Comment by lays147** [comment]
@gchristidis I described that scenario in my first comment, and I wasn't getting any diff. But I tested again, and now I get a diff: 
```
deleting route mock
deleting service mock
Summary:
  Created: 0
  Updated: 0
  Deleted: 2
```
Idk what is different between these tests(maybe I didn't pay enough attention, anyway).

However, I think that it should be made clear in Kong docs how this works. Most of us use the declarative config, and the docs are focused on calling the admin api directly. 

I believe that the community would favor in a full documentation of how to use the declarative configuration fully.
